### PR TITLE
MNIST CNN example

### PR DESCRIPTION
**Pull request by LazaroHurtado**
Introduced a new example file showcasing our framework's ability to load pre-trained Pytorch models and use convolutional layers. The model found in this example has a 96% accuracy and was taken from a public Kaggle [notebook](https://www.kaggle.com/code/anubhavgoyal10/pytorch-tutorial), the only change being the Softmax activation function so I can work with probabilities as opposed to logits. A bash script was also included to download the MNIST dataset file when the example is run rather than making it part of our repo.

### CODE REVIEW

File: examples/mnist_cnn.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+fn main() {
+    Command::new("/bin/sh")
+        .arg("-c")
+        .arg("./examples/download_mnist_dataset.sh")
+        .output()
+        .expect("Could not download MNIST dataset:");
+
+    let mnist = Box::new(
+        MnistBuilder::new()
```
Comment (jrego43) [inline examples/mnist_cnn.rs:37]: I'm clearly missing something, but where does MnistBuilder come from?

**Review by jrego43 — APPROVED** [review]
LGTM, great to finally have MNIST working!

**Review by Demba3 — APPROVED** [review]
Good job 🚢🚢
